packages/resolver/lib/sources/globalnpm.ts

```diff
@@ -10,41 +10,58 @@ export class GlobalNPM implements ResolverSource {
     if (importPath.indexOf(".") === 0 || path.isAbsolute(importPath)) {
       return null;
     }
-    const contract_name = path.basename(importPath, ".sol");
+    const contractName = path.basename(importPath, ".sol");
 
-    let [package_name] = importPath.split("/", 1);
-    if (detectInstalled.sync(package_name)) {
-      const regex = new RegExp(`/${package_name}$`);
-      const global_package_path = get_installed_path
-        .getInstalledPathSync(package_name)
+    let [packageName] = importPath.split("/", 1);
+    if (detectInstalled.sync(packageName)) {
+      const regex = new RegExp(`/${packageName}$`);
+      const globalPackagePath = getInstalledPath
+        .getInstalledPathSync(packageName)
         .replace(regex, "");
-      const expected_path = path.join(
-        global_package_path,
-        package_name,
-        "build",
-        "contracts",
-        contract_name + ".json"
+
+      const result = this.resolveAndParse(
+        globalPackagePath,
+        packageName,
+        contractName
+      );
+      // result is null if it fails to resolve
+      if (result) {
+        return result;
+      }
+    }
+    return null;
+  }
+
+  resolveAndParse(basePath: string, packageName: string, contractName: string) {
+    const packagePath = path.join(basePath, packageName);
+    const subDirs = [`build${path.sep}contracts`, "build"];
+    for (const subDir of subDirs) {
+      const possiblePath = path.join(
+        packagePath,
+        subDir,
+        `${contractName}.json`
       );
       try {
-        const result = fs.readFileSync(expected_path, "utf8");
+        const result = fs.readFileSync(possiblePath, "utf8");
```

Comment:
Is the intention to determine if the possible path exists and then parse it? It seems weird to use the exception handler to continue the loop.

Also, what happens if there are multiple paths exists? That is a consumable file in build, as well as contracts folder?

Reply:
Basically the npm and global npm resolvers have two locations that they search in the package: `/build/contracts` and `/build`. It will look in those locations in that order and use the first artifact it finds.

I would kind of agree with you about using the error handling but this is in fact what Node recommends. Check out the docs for [`stat`](https://nodejs.org/api/fs.html#fs_fs_stat_path_options_callback).

Reply:
TIL node's weird try/catch recommendation. Thanks @eggplantzzz! Is there's documentation to explain the path ordering/precedence?

Reply:
Are you talking about documentation about the resolver? Absolutely not :)